### DISCUSSION TITLE
Full-fledged Intellisense and embedded polyglot python integration support?

### DISCUSSION THREAD

**Discussion by s3ts** [oracle/graalpython#416] [Q&A] [answered]
The big advantage for me to use graalvm.python is to embed python coding abilities into an existing Java API.

However, it seems you had a VS Code plugin that helped debug anywhere into Java/GraalPy, but you deprecated this to almost single-out GraalVM implementation of Python and GraalVM itself. 

I mean, we can live off of "inspect" feature you've added to graalpy, but a full-featured LSP would take so much pain away from this being a real option. Seems it existed, what happened there?

I can live with incompatibilities with existing python APIs that already work with CPython. What can we expect moving forward to support the integrated pathway with GraalVM and the polyglot python implementation?

**Comment by MartinBalin** [comment, marked as answer]
If you mean former GraalVM Extension which provided these features then it is no longer supported and was unpublished from VSCode Marketplace.

**Reply by s3ts** [reply]
Thank you for an answer here, but can we still know what happened to the LSP/Debugger for GraalVM-Python? I would think the dual-capability of debugging Java and Python at the same time would be a great feature to preserve. Whats the reason behind removing it?

**Reply by fniephaus** [reply]
Just to be clear: LSP and debugging support are two different features, and none of them have been removed.

What we have removed, because it was no longer maintained for some time, is the GraalVM Extension for VS Code, which used NetBeans under the hood to provide Java<->Python debugging. While we unfortunately had to remove this integration, we do want to provide Java<->Python debugging again in the future, probably based on the DAP and/or in IntelliJ IDEA.
